tests/NHSD.BuyingCatalogue.Ordering.Services.UnitTests/DefaultDeliveryDateServiceTests.cs

```diff
@@ -82,15 +83,38 @@ public static void Constructor_NullHttpContextAccessor_ThrowsArgumentNullExcepti
             Order order,
             DefaultDeliveryDateService service)
         {
-            order.SetDefaultDeliveryDate(catalogueItemId, defaultDeliveryDate);
-            context.Add(order);
+            context.Order.Add(order);
             await context.SaveChangesAsync();
+            order.SetDefaultDeliveryDate(catalogueItemId, defaultDeliveryDate);
 
             DeliveryDateResult result = await service.SetDefaultDeliveryDate(callOffId, catalogueItemId, defaultDeliveryDate.AddDays(-1));
 
             result.Should().Be(DeliveryDateResult.Updated);
         }
 
+        [Test]
+        [InMemoryDbAutoData]
+        public static async Task SetDefaultDeliveryDate_SavesToDb(
+            [Frozen] ApplicationDbContext context,
+            [Frozen] CallOffId callOffId,
+            [Frozen] CatalogueItemId catalogueItemId,
+            DateTime defaultDeliveryDate,
+            Order order,
+            DefaultDeliveryDateService service)
+        {
+            context.Order.Add(order);
+            await context.SaveChangesAsync();
+
+            DeliveryDateResult result = await service.SetDefaultDeliveryDate(callOffId, catalogueItemId, defaultDeliveryDate.AddDays(-1));
```

Comment:
`result` isn't used so we can remove the assignment.

Reply:
Removed `result` assignment